### PR TITLE
Use common scripting for Docker in GitHub actions

### PR DESCRIPTION
**Pull request by GMNGeoffrey**

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
Based on #9841. Only the last commit is part of this PR

File: build_tools/github_actions/docker_run.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 The IREE Authors
```
Comment (ScottTodd) [inline build_tools/github_actions/docker_run.sh:1]: 2022

File: build_tools/github_actions/docker_run.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Runs docker configured for usage with Kokoro, translating Kokoro-specific
+# environment variables into generic ones and then invoking `docker run` with
+# some arguments to make it behave itself.
```
Comment (ScottTodd) [inline build_tools/github_actions/docker_run.sh:9]: update comment (GitHub Actions, not Kokoro)

**Review by ScottTodd — APPROVED** [review]
lgtm once comments are addressed and the base PR is merged

File: build_tools/github_actions/docker_run.sh (hunk 1 above)
Comment (GMNGeoffrey) [reply build_tools/github_actions/docker_run.sh:1]: Forked :-P

File: build_tools/github_actions/docker_run.sh (hunk 2 above)
Comment (GMNGeoffrey) [reply build_tools/github_actions/docker_run.sh:9]: Done

**Comment by GMNGeoffrey** [comment]
Forgot to tag originally, but part of https://github.com/iree-org/iree/issues/9855

**Pull request merged**